pkg/cvo/availableupdates_test.go

```diff
@@ -0,0 +1,306 @@
+package cvo
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/cluster-version-operator/pkg/clusterconditions"
+)
+
+type queueStub struct{}
+
+func (q queueStub) Add(interface{}) {}
+
+func (q queueStub) Len() int {
+	panic("implement me")
+}
+
+func (q queueStub) Get() (interface{}, bool) {
+	panic("implement me")
+}
+
+func (q queueStub) Done(interface{}) {
+	panic("implement me")
+}
+
+func (q queueStub) ShutDown() {
+	panic("implement me")
+}
+
+func (q queueStub) ShutDownWithDrain() {
+	panic("implement me")
+}
+
+func (q queueStub) ShuttingDown() bool {
+	panic("implement me")
+}
+
+func (q queueStub) AddAfter(interface{}, time.Duration) {
+	panic("implement me")
+}
+
+func (q queueStub) AddRateLimited(interface{}) {
+	panic("implement me")
+}
+
+func (q queueStub) Forget(interface{}) {
+	panic("implement me")
+}
+
+func (q queueStub) NumRequeues(interface{}) int {
+	panic("implement me")
+}
+
+// notFoundProxyLister is a stub for ProxyLister
+type notFoundProxyLister struct{}
+
+func (n notFoundProxyLister) List(labels.Selector) ([]*configv1.Proxy, error) {
+	return nil, nil
+}
+
+func (n notFoundProxyLister) Get(name string) (*configv1.Proxy, error) {
+	return nil, errors.NewNotFound(schema.GroupResource{Group: configv1.GroupName, Resource: "proxy"}, name)
+}
+
+type notFoundConfigMapLister struct{}
+
+func (n notFoundConfigMapLister) List(labels.Selector) ([]*corev1.ConfigMap, error) {
+	return nil, nil
+}
+
+func (n notFoundConfigMapLister) Get(name string) (*corev1.ConfigMap, error) {
+	return nil, errors.NewNotFound(schema.GroupResource{Group: "", Resource: "configmap"}, name)
+}
+
+type fakeConditionRegistry struct{}
+
+func (f fakeConditionRegistry) Register(string, clusterconditions.Condition) {
+	panic("implement me")
+}
+
+func (f fakeConditionRegistry) PruneInvalid(ctx context.Context, matchingRules []configv1.ClusterCondition) ([]configv1.ClusterCondition, error) {
+	return matchingRules, nil
```

Comment:
nit: instead of declaring a fake condition registry, you could possibly use [the mock package][1]:

```go
mock := mock.Mock{}
conditionRegistry := clusterconditions.NewConditionRegistry()
conditionRegistry.Register("Always", &always.Always{})
conditionRegistry.Register("PromQL", &mock)
// before calling with PromQL, load up mock.ValidQueue and mock.MatchQueue with the return values you  want
```

or  some such.

[1]: https://github.com/openshift/cluster-version-operator/blob/53a5d57c2bd3c92c145dac005fab6e4cdcf8f43b/pkg/clusterconditions/mock/mock.go#L34-L44

Reply:
I have missed that we have this mock! I'll definitely want to use it instead of adding new code.